File: quiz_app_flutter/lib/domain/usecase/quiz_set/getquizsetlist_usecase.dart
import 'package:quiz_app_flutter/core/resource.dart';
import 'package:quiz_app_flutter/domain/entities/quiz_set/quiz_set_data.dart';
import 'package:quiz_app_flutter/domain/repository/quiz_repository.dart';

class GetQuizSetListUseCase {
  final QuizRepository repository;

  GetQuizSetListUseCase(this.repository);

  Stream<Resource<QuizSetData>> call(String topic) {
    return repository.getQuizListByTopic(topic);
  }
}

File: quiz_app_kmp/composeApp/src/commonMain/kotlin/com/tnm/quizappkmp/domain/usecase/quizSets/GetQuizSetListUseCase.kt
package com.tnm.quizappkmp.domain.usecase.quizSets

import com.tnm.quizappkmp.domain.model.quizSets.QuizSetData
import com.tnm.quizappkmp.domain.repositry.QuizRepository
import kotlinx.coroutines.flow.Flow

class GetQuizSetListUseCase(
    private val repository: QuizRepository
) {
    operator fun invoke(topic: String): Flow<QuizSetData?> {
        return repository.getQuizListByTopic(topic)
    }
}